semana10/projeto-future-x/src/containers/AdminPage/index.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { push, replace } from 'connected-react-router';
+import { routes } from '../Router';
+import { Button, Typography } from '@material-ui/core';
+import styled from 'styled-components';
+
+const ContainerAdmin = styled.div `
+    display: flex;
+    flex-direction: column;
+    margin: 0 35rem;
+    gap: 5px;
+    text-align: center;
+`
+
+class AdminPage extends React.Component{
+
+    componentDidMount() {
+        const token = localStorage.getItem('token')
+        if(token === null) {
+            this.props.goToLoginPage()
+        }
+    }
+
+    handleLogout = () => { 
+        localStorage.clear();
+        window.location.reload()
```

Comment:
Aqui, ao invés de atualizar a página, vc poderia ter chamado a função `this.props.goToLoginPage()`.